qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/TypeRemovalIT.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    private void createTemplate(String templateName, String indexPattern, String mapping) throws IOException {
+        Request templateRequest = new Request(HttpPut.METHOD_NAME, "/_template/" + templateName);
+        String entity = "{\"index_patterns\": \"" + indexPattern + "\"";
+        if (mapping != null) {
+            entity += ",\"mappings\" : {" + mapping + "}";
+        }
+        entity += "}";
+        templateRequest.addParameter("include_type_name", "true");
+        templateRequest.setJsonEntity(entity);
+        useIgnoreTypesRemovalWarningsHandler(templateRequest);
+        client().performRequest(templateRequest);
+    }
+
+    private void reindex(String originalIndex, String newIndex) throws IOException {
+        Request reIndexRequest = new Request(HttpPost.METHOD_NAME, "/_reindex/");
+        String entity = "{ \"source\": { \"index\": \"" + originalIndex + "\" }, \"dest\": { \"index\": \"" + newIndex + "\" } }";
+        reIndexRequest.setJsonEntity(entity);
+        reIndexRequest.addParameter("refresh", "true");
+        client().performRequest(reIndexRequest);
+    }
+
+    private void assertIndexMapping(String index, String mappings) throws IOException {
+        Request testIndexMappingRequest = new Request(HttpGet.METHOD_NAME, "/" + index + "/_mapping");
+        Response testIndexMappingResponse = client().performRequest(testIndexMappingRequest);
+        assertEquals("{\""+index+"\":{\"mappings\":{"+mappings+"}}}",
+            EntityUtils.toString(testIndexMappingResponse.getEntity(), StandardCharsets.UTF_8));
+    }
+
+    private void assertCount(String index, int count, Integer totalShards) throws IOException {
+        Request searchTestIndexRequest = new Request(HttpPost.METHOD_NAME, "/" + index + "/_search");
+        searchTestIndexRequest.addParameter(TOTAL_HITS_AS_INT_PARAM, "true");
+        if (totalShards != null) {
+            searchTestIndexRequest.addParameter("filter_path", "hits.total,_shards");
+        } else {
+            searchTestIndexRequest.addParameter("filter_path", "hits.total");
+        }
+
+        Response searchTestIndexResponse = client().performRequest(searchTestIndexRequest);
+        String expectedResponse;
+        if (totalShards != null) {
+            expectedResponse = "{\"_shards\":{\"total\":" + totalShards + ",\"successful\":" + totalShards + ",\"skipped\":0,\"failed\":0},\"hits\":{\"total\":" + count + "}}";
+        } else {
+            expectedResponse = "{\"hits\":{\"total\":" + count + "}}";
+        }
+        assertEquals(expectedResponse,
+            EntityUtils.toString(searchTestIndexResponse.getEntity(), StandardCharsets.UTF_8));
+    }
+
+    private final Pattern TYPE_REMOVAL_WARNING = Pattern.compile(
+        "^\\[types removal\\] (.+) include_type_name (.+) is deprecated\\. The parameter will be removed in the next major version\\.$"
+    );
+
+    private void useIgnoreTypesRemovalWarningsHandler(Request request) {
```

Comment:
I'm curious is this new method having a same function with the existing method https://github.com/opensearch-project/OpenSearch/blob/1.3.1/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L326 ?